WcaOnRails/app/webpacker/components/Translators/index.jsx

```diff
@@ -0,0 +1,44 @@
+import React, { useMemo } from 'react';
+import { Container, Grid, Header } from 'semantic-ui-react';
+import useLoadedData from '../../lib/hooks/useLoadedData';
+import { apiV0Urls } from '../../lib/requests/routes.js.erb';
+import { groupTypes } from '../../lib/wca-data.js.erb';
+import Loading from '../Requests/Loading';
+import Errored from '../Requests/Errored';
+import I18n from '../../lib/i18n';
+import UserBadge from '../UserBadge';
+
+export default function Translators() {
+  const {
+    data: translators, loading, error,
+  } = useLoadedData(apiV0Urls.userRoles.listOfGroupType(groupTypes.translators));
+  const groupedTranslators = useMemo(
+    () => Object.groupBy(translators || [], (role) => role.group.id),
```

Comment:
Oh I didn't knew about it. In that case, will it make sense to use `Array.reduce` here?

The reason why I didn't use this because I wanted the sort order to be maintained there, but it was not so necessary here because all the translation groups are of equal priority. But thinking again, this will lead to inconsistency, I should maintain the order as well and I guess I have to do Array.reduce along with some other array to store order. Thoughts?

Reply:
Looks like you already used `Object.groupBy` in the `RegionManager`.

I feel like calling `groupBy` and then sorting the (sub)-results as necessary would be a shorter/easier-to-understand approach, since it makes the 2 main requirements very explicit and the arrays aren't long enough for efficiency to be a major factor. I wasn't thinking about order when I was reading this code.

If we don't want to use `groupBy` due to not working on safari (@gregorbg?) then defining our own util function `groupBy` might be best, since it's being used in at least 3 different places.

Reply:
I just realized we have `groupBy` in lodash. I used that instead, hope this is fine now.

Reply:
I didn't change the `groupBy` in RegionManager as it is not related to this PR. I've noted it down in my notes, will handle it once I make some change in RegionManager (which I probably will sometime soon).